drivers/adc/ad7124/ad7124.c

```diff
@@ -212,10 +213,32 @@ int32_t ad7124_read_register(struct ad7124_dev *dev,
 	return ret;
 }
 
+/**
+ * @brief Wrap the read register function to give it a modern signature.
+ * @param [in] dev - Driver handler pointer.
+ * @param [in] reg - Address of the register to be read.
+ * @param [out] readval - Pointer to the register value.
+ * @return SUCCESS in case of success, error code otherwise.
+ */
+int32_t ad7124_read_register2(struct ad7124_dev *dev, uint32_t reg,
+			      uint32_t *readval)
+{
+	int32_t ret;
+
+	ret = ad7124_read_register(dev, &dev->regs[reg]);
+	if (ret != 0)
+		return ret;
+
+	*readval = dev->regs[reg].value;
+
+	return 0;
```

Comment:
`return ret`?
Check throughout the entire code.

Reply:
will do